templates/server/web/app_test.go

```diff
@@ -0,0 +1,89 @@
+package web
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"project-name/model"
+	"testing"
+)
+
+var expectedTechnologies = []*model.Technology{
+	{
+		Name:    "Tech1",
+		Details: "Details1",
+	},
+	{
+		Name:    "Tech2",
+		Details: "Details2",
+	},
+}
+
+type MockDb struct {
+	called     int
```

Comment:
I think checking how many times the method was called is not required. First of all, it is only a template and the user might not want to check this. Second, it is a debatable point if this check should be applied or not. Say, if it is applied and we expect someone implementing this function wrongly by calling the db several times we assume that we have no access to the code and we apply "black box" testing. Which is obviously not the case because it is a unit test. Does that sound fine to you?

Reply:
I agree with keeping the project template as unopinionated as possible, therefore i removed the check